appscale/tools/remote_helper.py

```diff
@@ -95,6 +91,55 @@ class RemoteHelper(object):
 
 
   @classmethod
+  def start_all_nodes(cls, options, count):
+    """ Starts all nodes in the designated public cloud.
+
+    Args:
+      options: A Namespace that includes parameters passed in by the user that
+        define non-placement-strategy-related deployment options (e.g., keypair
+        names, security group names).
+      count: A int, the number of nodes to spawn.
+    Returns:
+      The public IPs and instance IDs (dummy values in non-cloud deployments)
+      corresponding to the nodes that were started.
+    """
+    if not options.infrastructure:
+      return [], [], []
+
+    instance_ids, public_ips, private_ips = cls.spawn_nodes_in_cloud(
+      options, count=count)
+
+    for instance_id, public_ip in zip(instance_ids, public_ips):
+      try:
+        cls.ensure_machine_is_compatible(public_ip, options.keyname,
+          options.table, options.verbose)
+      except AppScaleException as ase:
+        # On failure shutdown the cloud instances, cleanup the keys, but only
+        # if --test is not set.
+        if options.infrastructure:
+          if not options.test:
+            try:
```

Comment:
We only terminate them if we're not in testing mode? I guess that sort of makes sense, but it could be surprising since we don't really use test that way anywhere else.